src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php

```diff
@@ -783,18 +783,36 @@ public function prepare_item_for_response( $item, $request ) {
 					}
 
 					$size_data['source_url'] = $image_src[0];
+
+					if ( empty( $size_data['sources'] ) || ! is_array( $size_data['sources'] ) ) {
+						continue;
+					}
+
+					$directory = dirname( $size_data['source_url'] );
+					foreach ( $size_data['sources'] as $mime => &$mime_details ) {
+						$source_url                 = "{$directory}/{$mime_details['file']}";
+						$mime_details['source_url'] = $source_url;
+					}
 				}
 
 				$full_src = wp_get_attachment_image_src( $post->ID, 'full' );
 
 				if ( ! empty( $full_src ) ) {
+					$directory = dirname( $full_src[0] );
+					foreach ( $data['media_details']['sources'] as $mime => &$mime_details ) {
+						$source_url                 = "{$directory}/{$mime_details['file']}";
+						$mime_details['source_url'] = $source_url;
+					}
+
 					$data['media_details']['sizes']['full'] = array(
 						'file'       => wp_basename( $full_src[0] ),
 						'width'      => $full_src[1],
 						'height'     => $full_src[2],
 						'mime_type'  => $post->post_mime_type,
 						'source_url' => $full_src[0],
+						'sources'    => $data['media_details']['sources'],
```

Comment:
See above, this should only be set here if `$data['media_details']['sources']` is actually set. You can combine that all in one if clause.

Reply:
Thanks. This feedback is addressed.